app/services/katello/pulp3/api/core.rb

```diff
@@ -192,6 +200,12 @@ def delete_distribution(href)
           ignore_404_exception { distributions_api.delete(href) }
         end
 
+        def core_repositories_list_all(options = {})
+          self.class.fetch_from_list do |page_opts|
+            core_repositories_api.list(page_opts.merge(options))
+          end
+        end
```

Comment:
didn't  an issue get filed to make the cleanup default to all repositories?  did we open a katello issue to track that too?

Reply:
What do you mean by default to all repositories? You can clean all repos for any smart proxy by doing it from their UI pages.

Reply:
Oh I remember what you mean now, the ability to not need to provide any repository hrefs.  I'll make an issue.

Reply:
Redmine: https://projects.theforeman.org/issues/34080